recipes-core/initrdscripts/files/ptest/test_ni_provisioning.sh

```diff
@@ -0,0 +1,176 @@
+		found_error=true
+	fi
+}
+
+
+found_error=false
+
+if uname --machine | grep -q arm; then
+	echo "INFO: detected an ARM machine architecture." \
+	     "This ptest is only valid for non-ARM, RAUC systems."
+	ptest_skip
+fi
+
+
+## PARTITION TESTS ##
+echo "## Checking that RAUC partitons are present..."
+# The UUIDs here should match those set by the disk_config_x64 file.
+# https://github.com/ni/meta-nilrt/blob/983d50796302b394aa428b992dae48b4da32ff08/recipes-core/initrdscripts/files/disk_config_x64#L28
+RE_ESP_UUID=C12A7328\-F81F\-11D2\-BA4B\-00A0C93EC93B
+RE_X64_ROOT_UUID=4F68BCE3\-E8CD\-4DB1\-96E7\-FBCAF984B709
+
+# Determine the configured /boot storage device. Assert that this device should
+# be the root of the partitions we evaluate in the remainder of this test.
+# For RAUC systems, we expect this to be: /dev/sda, but it isn't required to be.
+boot_part=$(findmnt --kernel --first-only --noheadings --output SOURCE /boot)
+test -n "${boot_part}" || (print_error "No detected /boot mount; system state invalid."; ptest_fail)
+boot_dev=/dev/$(lsblk -no PKNAME ${boot_part})
+test -e "${boot_dev}" || (print_error "Boot device ${boot_dev} does not exist; system state invalid."; ptest_fail)
+
+
+readarray -t parts < <(fdisk -l ${boot_dev} -o Device,Name,Type-UUID | grep -o -E '^\/dev\/.*')
+
+# ASSERT: part 0 is: niboota and is an ESP partition
+test_part_id ${parts[0]} niboota $RE_ESP_UUID
+# ASSERT: part 1 is: nibootb and is an ESP partition
+test_part_id ${parts[1]} nibootb $RE_ESP_UUID
+# ASSERT: part 2 is: niuser and is any partition type
+test_part_id ${parts[2]} niuser $RE_X64_ROOT_UUID
+
+# bail out if the partition tests failed
+$found_error && ptest_fail
+
+
+## RAUC PARTITION DEVICE LINKS TESTING ##
+echo "## Checking for RAUC partition device links in /dev..."
+test -d /dev/niboot || print_error "/dev/niboot does not exist."
+
+niboot_links=(\
+	/dev/niboot/niboot.current \
+	/dev/niboot/niboot.other \
+	/dev/niboot/niboota \
+	/dev/niboot/nibootb \
+	/dev/niboot/niuser \
+)
+for link in ${niboot_links[@]}; do
+	if [ -L "$link" ]; then
+		echo "${link} [OK]"
+	else
+		echo "${link} [FAIL]"
+		print_error "Device link \"$link\" does not exist."
+	fi
+done
+# the link subtests are not critical failures, so keep testing, even if they fail.
+
+
+## PARTITION CONTENTS TESTING ##
+echo "## Checking that niboot* contents are similar..."
+set -eo pipefail
+
+# Unmount the read-only partitions and remove their empty directories
+unmount_ro_partitions () {
+	set +e
+	for part in $@; do
+		[ -d ${part} ] && umount -f ${part} && rmdir ${part}
+	done
+}
+
+# After provisioning, niboota and nibootb should contain exactly the same contents.
+mnt_a=$(mktemp -d --tmpdir $PTEST_TEST.niboota.XXXXX)
+mnt_b=$(mktemp -d --tmpdir $PTEST_TEST.nibootb.XXXXX)
+
+trap "unmount_ro_partitions ${mnt_a} ${mnt_b}; exit 99" EXIT
+
+mount -o ro -L niboota ${mnt_a}
+mount -o ro -L nibootb ${mnt_b}
```

Comment:
If you don't want to check the /dev/niboot/* links, then lets at least change this code to mount /dev/niboot/niboota and /dev/niboot/nibootb.